File: app/src/main/java/com/wade/mil/Button.java
package com.wade.mil;

import java.util.ArrayList;
import java.util.List;

public interface Button {
    String name = "Button";
    List<Field> field = new ArrayList<>();
    int step = -1;
    int mode = Const.MODE_AUTO;
    double d=0.0f, a=0.0f;
    void setName(String n);
    void addField(String n, String d, boolean a, int m);
    void setMode(int m);
    String getDesc();
    void setValue(String v);
    String getValue();
    //    double getDegree() {
//        return field[step].degree;
//    }
    boolean next();
    boolean back();
    double[] calc();
    String string();
}

File: app/src/main/java/com/wade/mil/Button.kt
package com.wade.mil

interface Button {
    fun setName(n: String?)
    fun addField(n: String?, d: String?, a: Boolean, m: Int)
    fun setMode(m: Int)
    val desc: String?
    var value: String?

    //    double getDegree() {
    //        return field[step].degree;
    //    }
    operator fun next(): Boolean
    fun back(): Boolean
    fun calc(): DoubleArray?
    fun string(): String?

    companion object {
        const val name = "Button"
        val field: List<Field> = ArrayList()
        const val step = -1
        const val mode = Const.MODE_AUTO
        const val d = 0.0
        const val a = 0.0
    }
}